src/main/java/com/conveyal/r5/streets/StreetLayer.java

```diff
@@ -1207,6 +1207,9 @@ public int getOrCreateVertexNear(double lat, double lon, StreetMode streetMode)
         EdgeStore.Edge newEdge1 = edgeStore.addStreetPair(newVertexIndex, oldToVertex, split.distance1_mm, edge.getOSMID());
         // Copy the flags and speeds for both directions, making newEdge1 like the existing edge.
         newEdge1.copyPairFlagsAndSpeeds(edge);
+
+        if (geoms.size() > 1) newEdge1.setGeometry(geoms.get(1));
```

Comment:
I think `geoms` should always be defined, and always have length 2 (hence my suggestion to use a P2 type above). If in some branch the geometry split would not happen, I would expect an uninitialized value of `geoms == null`.